File: ios/Classes/Clustering/TiPOIItem.m
/**
 * Ti.GoogleMaps
 * Copyright (c) 2015-Present by Hans Knoechel, Inc. All Rights Reserved.
 * Licensed under the terms of the Apache Public License
 * Please see the LICENSE included with this distribution for details.
 */

#import "TiPOIItem.h"

@implementation TiPOIItem

- (instancetype)initWithPosition:(CLLocationCoordinate2D)position andTitle:(NSString *)title subtitle:(NSString *)subtitle icon:(UIImage *)icon userData:(NSDictionary *)userData;
{
    if (self = [super init]) {
        _position = position;
        _title = title;
        _subtitle = subtitle;
        _icon = icon;
        _userData = userData;
    }
    return self;
}

@end

File: titanium-googlemaps-swift/Classes/TiPOIItem.swift
//
//  TiPOIItem.swift
//  TiGooglemaps
//
//  Created by Hans Knöchel on 26.10.25.
//

import Foundation
import CoreLocation
import UIKit
import GoogleMapsUtils

@objcMembers
class TiPOIItem: NSObject, GMUClusterItem {
  let position: CLLocationCoordinate2D
  var title: String?
  let subtitle: String?
  let icon: UIImage?
  let userData: [AnyHashable: Any]?
  
  @objc(initWithPosition:andTitle:subtitle:icon:userData:)
  init(position: CLLocationCoordinate2D,
       andTitle title: String?,
       subtitle: String?,
       icon: UIImage?,
       userData: [AnyHashable: Any]?) {
    self.position = position
    self.title = title
    self.subtitle = subtitle
    self.icon = icon
    self.userData = userData
    
    super.init()
  }
}
